pkg/controllers/termination/terminate.go

```diff
@@ -122,3 +115,32 @@ func (t *Terminator) getPods(ctx context.Context, node *v1.Node) ([]*v1.Pod, err
 	}
 	return ptr.PodListToSlice(pods), nil
 }
+
+func (t *Terminator) getEvictablePods(pods []*v1.Pod) []*v1.Pod {
+	evictable := []*v1.Pod{}
+	for _, p := range pods {
+		if scheduling.Tolerates(p, v1.Taint{Key: v1.TaintNodeUnschedulable, Effect: v1.TaintEffectNoSchedule}) == nil {
+			continue
+		}
+		evictable = append(evictable, p)
+	}
+	return evictable
+}
+
+func (t *Terminator) evict(ctx context.Context, pods []*v1.Pod) {
+	// 1. Prioritize noncritical pods https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
+	prioritized := []*v1.Pod{}
+	for _, p := range pods {
+		if !p.DeletionTimestamp.IsZero() {
```

Comment:
IIUC, these pods will be included on line 143. Filtering these before they even enter this context might save you. Maybe most easily done on line 131.